server/internal/testutil/filesystem.go

```diff
@@ -0,0 +1,42 @@
+package testutil
+
+import (
+	"context"
+	"os"
+	"testing"
+)
+
+// / WithTempDir creates a tempdir using the std testing lib
+func WithTempDir() IntegrationDependency {
+	return IntegrationDependency(func(ctx context.Context, t *testing.T) (context.Context, func()) {
+		t.Helper()
+
+		dir, err := os.MkdirTemp("", "")
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		ctx = context.WithValue(ctx, dirKey{}, dir)
+		return ctx, func() {
+			if err := os.RemoveAll(dir); err != nil {
+				t.Fatal(err)
+			}
+		}
+	})
+}
+
+type dirKey struct{}
+
+// / TempDirFromContext returns the path to the generated tempdir
+func TempDirFromContext(ctx context.Context) string {
+	dir, ok := ctx.Value(dirKey{}).(string)
+	if !ok {
+		panic("failed to cast tempdir from context to string")
```

Comment:
🥜 What about using `t.Fatal`? This this function is intended to be used only during tests

Reply:
ye, I didn't want to add `t` as an arg but I guess the UX is better